wcfsetup/install/files/lib/util/ImageUtil.class.php

```diff
@@ -180,6 +180,23 @@ public static function fixOrientation($filename) {
 		return $filename;
 	}
 	
+	/**
+	 * Examines the `accept` header to determine if the browser
+	 * supports WebP images.
+	 */
+	public static function browserSupportsWebP(): bool {
+		static $supportsWebP = null;
+		
+		if ($supportsWebP === null) {
+			$supportsWebP = false;
+			if (!empty($_SERVER["HTTP_ACCEPT"]) && preg_match("~(?:^|[,;])image/webp[,;]~", $_SERVER["HTTP_ACCEPT"])) {
```

Comment:
This misparses the `HTTP_ACCEPT` header. It's not dangerous in this case, but I'd rather have this correct:

1. `explode()` HTTP_ACCEPT at commas.
2. Explode each array item at semicolons.
3. Trim the resulting values.
4. If the first item is `image/webp` then webp is supported.

Something like this should work. It can be simplified a bit (e.g. replace the ArrayUtil::trim() by a trim() on the mimeType).

```php
$acceptableMimeTypes = array_map(function ($acceptableMimeType) {
 [$mimeType] = ArrayUtil::trim(explode(';', $acceptableMimeType));
 return $mimeType;
}, explode(',', $_SERVER['HTTP_ACCEPT']));
return in_array('image/webp', $acceptableMimeTypes);
```